### PR TITLE
Report error for trying to create a key through API if it exists

### PR DESCRIPTION
**Pull request by bdonald25**
#1988

### CODE REVIEW

**Comment by jaypoulz** [comment]
It looks like your failing your own test :/

**Comment by bdonald25** [comment]
Yeah, so my tests don't work locally. I have not been able to get them working correctly since my laptop was wiped so I have to let travis test them for me, just until I can figure out what is wrong.

**Comment by jaypoulz** [comment]
```
def add_key
        if params[:contrib_key].nil?
          respond_to do |format|
            format.json { render json: { msg: 'contrib_key is nil' }, status: :unprocessable_entity }
          end
        return
        end
        
        if params[:contrib_key][:name].nil?
          respond_to do |format|
            format.json { render json: { msg: 'name is nil' }, status: :unprocessable_entity }
          end
        return
        end
        
        if params[:id].nil?
          respond_to do |format|
            format.json { render json: { msg: 'project id is nil' }, status: :unprocessable_entity }
          end
          return
        end

        if params[:contrib_key][:key].nil?
          respond_to do |format|
            format.json { render json: { msg: 'key is nil' }, status: :unprocessable_entity }
          end
          return
        end

          @project = Project.find(params[:id])
          unless @cur_user.id == @project.user_id
           respond_to do |format|
              format.json { render json: { msg: 'User does not own the project.' }, status: :unauthorized }
            end
           return
           end

            key = @project.contrib_keys.find_by_key(params[:contrib_key][:key])
            unless key.nil?
              respond_to do |format|
                format.json { render json: { msg: 'key already exists.' }, status: :conflict }
              end
              return
            end

             key = ContribKey.new(name: params[:contrib_key][:name], project_id: @project.id, key: params[:contrib_key][:key])
              respond_to do |format|
                if key.save
                  format.json { render json: { msg: 'Success' }, status: :created }
                else
                  format.json { render json: { msg: 'Unprocessable Entity.' }, status: :unprocessable_entity }
                end
              end
          end
        end
      end
    end
  end
end
```

File: app/controllers/api/v1/projects_controller.rb [hunk 1]
```diff
@@ -67,43 +67,55 @@ def add_key
           respond_to do |format|
             format.json { render json: { msg: 'contrib_key is nil' }, status: :unprocessable_entity }
           end
-        elsif params[:contrib_key][:name].nil?
+        return
```
Comment (jaypoulz) [inline app/controllers/api/v1/projects_controller.rb]: these returns should line up with the respond_to block

**Comment by jaypoulz** [comment]
Sirens! :cop:

**Pull request merged**